test/Parse/multiline_string.swift

```diff
@@ -0,0 +1,116 @@
+// RUN: %target-swift-frontend -dump-ast %s 2>&1 | %FileCheck %s
+
+import Swift
+
+// ===---------- Multiline --------===
+
+_ = """
+    One
+    ""Alpha""
+    """
+// CHECK: "One\n\"\"Alpha\"\""
+
+_ = """
+    Two
+  Beta
+  """
+// CHECK: "  Two\nBeta"
+
+_ = """
+    Three
+    Gamma
+  """
+// CHECK: "  Three\n  Gamma"
+
+_ = """
+    Four
+    Delta
+"""
+// CHECK: "    Four\n    Delta"
+
+_ = """
+    Five\n
+
+    Epsilon
+    """
+// CHECK: "Five\n\n\nEpsilon"
+
+
+_ = """
+    Six
+    Zeta
+
+    """
+// CHECK: "Six\nZeta\n"
+
+_ = """
+    Seven
+    Eta\n
+    """
+// CHECK: "Seven\nEta\n"
+
+_ = """
+    \"""
+    "\""
+    ""\"
+    Iota
+    """
+// CHECK: "\"\"\"\n\"\"\"\n\"\"\"\nIota"
+
+_ = """
+     \("Nine")
+    Kappa
+    """
+// CHECK: "\nKappa"
+
+_ = """
+\\
+"""
+// CHECK: "\\"
+
+_ = """
+
+  ABC
+  """
+// CHECK: "\nABC"
+
+
+_ = """
+
+ABC
+"""
+// CHECK: "\nABC"
+
+_ = """
+  
+  ABC
+  """
+// CHECK: "\nABC"
+
+_ = """
+
+  ABC
+  """
+// CHECK: "\nABC"
+
+_ = """
+
+    ABC
+
+    """
+// CHECK: "\nABC\n"
+
+_ = """
+
+
+    """
+// CHECK: "\n"
+
+_ = """
+
+    """
+// CHECK: ""
+
+_ = """
+    """
+// CHECK: ""
```

Comment:
This is looking great, thanks!

Bear with me, I have two corner cases that I want to make sure gets tested. I also have a couple code nitpicks, but those are not necessarily blockers for this PR.

After those two corner cases, this will LGTM! Thank you so much for working on this!